src/main/panel/Game.java

```diff
@@ -143,14 +143,37 @@ private void collision() {
 
         if (r1.intersects(r2)) {
             // player1 hit the ball
-            float d = ((float) ball.getX() + (float) ball.getWidth() / 2f)
+            if(player1.ifSmash==true){  //check if smash
+                float d = ((float) ball.getX() + (float) ball.getWidth() / 2f)
                     - ((float) player1.getX() + (float) player1.getWidth() / 2f);
-            ball.hit(d, 1);
+                if(player1.ifJump == true&& d>0){
+                    ball.hit(d, true);
+                }else{
+                    ball.hit(d, false);
+                }
+                
+            }else{
+                float d = ((float) ball.getX() + (float) ball.getWidth() / 2f)
+                    - ((float) player1.getX() + (float) player1.getWidth() / 2f);
+
+                ball.hit(d, false);
+            }
```

Comment:
don't judge smash here, cause that will make code confusing, the main responsibility of this area is to check if  collision, you can judge if smash in `hit`